dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/savedSessionStore/SavedSessionStore.java

```diff
@@ -103,59 +105,19 @@ public static void saveSavedSessionStore(final SavedSessionStore sessionStore) t
     }
 
     public int addSession(final Session session) {
-        int index = 0;
-        if (sessions.size() == 0) {
-            this.sessions.add(new SavedSession(session.getSessionName(), session.getEndpoint(), session.getPortNo(), session.getProxyServer(),
-                    SavedCredentials.fromCredentials(session.getClient().getConnectionDetails().getCredentials()), session.getDefaultSession(), session.isUseSSL()));
-            index = 1;
-        } else if (containsSessionName(sessions, session.getSessionName())) {
-            final SavedSession savedSession = sessions.stream().filter(o -> o.getName().equals(session.getSessionName())).findFirst().get();
-            if (isNewValuePresent(savedSession, session)) {
-                index = sessions.indexOf(savedSession);
-                this.sessions.remove(savedSession);
-                this.sessions.add(index, new SavedSession(session.getSessionName(), session.getEndpoint(), session.getPortNo(), session.getProxyServer(),
-                        SavedCredentials.fromCredentials(session.getClient().getConnectionDetails().getCredentials()), session.getDefaultSession(), session.isUseSSL()));
-            } else {
-                return -1;
-            }
-
-        } else if (!containsSessionName(sessions, session.getSessionName())) {
-            this.sessions.add(new SavedSession(session.getSessionName(), session.getEndpoint(), session.getPortNo(), session.getProxyServer(),
-                    SavedCredentials.fromCredentials(session.getClient().getConnectionDetails().getCredentials()), session.getDefaultSession(), session.isUseSSL()));
-            index = sessions.size();
-        } else {
-            index = -2;
-        }
-        return index;
-    }
-
-    private static boolean isNewValuePresent(final SavedSession savedSession, final Session session) {
-        if (!savedSession.getName().equals(session.getSessionName()))
-            return true;
-        if (!savedSession.getCredentials().getAccessId().equals(session.getClient().getConnectionDetails().getCredentials().getClientId()))
-            return true;
-        if (!savedSession.getCredentials().getSecretKey().equals(session.getClient().getConnectionDetails().getCredentials().getKey()))
-            return true;
-        if (!savedSession.getEndpoint().equals(session.getEndpoint()))
-            return true;
-        if (!savedSession.getPortNo().equals(session.getPortNo()))
-            return true;
-        if (savedSession.getProxyServer() != null) {
-            if (!savedSession.getProxyServer().equals(session.getProxyServer())) {
-                return true;
-            }
-        } else if (session.getProxyServer() != null) {
-            return true;
+        final SavedSession saved = new SavedSession(session, SavedCredentials.fromCredentials(session.getClient().getConnectionDetails().getCredentials()));
+        if (!sessions.contains(saved)) {
+            this.sessions.add(saved);
```

Comment:
This is missing the case where we need to replace the SavedSession for a Session that has been updated.